arelle/plugin/validate/HMRC/__init__.py

```diff
@@ -161,15 +166,19 @@ def validateXbrlStart(val, parameters=None, *args, **kwargs):
         val.isAccounts = not val.isComputation
 
     val.txmyType = None
-    for doc in val.modelXbrl.modelDocument.referencesDocument:
-        ns = doc.targetNamespace
-        if ns:
-            if ns.startswith("http://www.xbrl.org/uk/char/"): val.txmyType = "charities"
-            elif ns.startswith("http://www.xbrl.org/uk/gaap/"): val.txmyType = "ukGAAP"
-            elif ns.startswith("http://www.xbrl.org/uk/ifrs/"): val.txmyType = "ukIFRS"
-            elif ns.startswith("http://xbrl.frc.org.uk/"): val.txmyType = "FRS"
-            else: continue
-            break
+    if any((concept.qname.namespaceURI.startswith(FRC_URL_DOMAIN) and concept.modelDocument.inDTS)
```

Comment:
Do charaities, ukGAAP, or ukIFRS overlap, or are they completely disjoint?  Wondering if the logic for those others should remain unchanged, e.g.:
```
    for doc in val.modelXbrl.modelDocument.referencesDocument:
        ns = doc.targetNamespace
        if ns:
            if ns.startswith("http://www.xbrl.org/uk/char/"): val.txmyType = "charities"
            elif ns.startswith("http://www.xbrl.org/uk/gaap/"): val.txmyType = "ukGAAP"
            elif ns.startswith("http://www.xbrl.org/uk/ifrs/"): val.txmyType = "ukIFRS"
            elif ns.startswith("http://xbrl.frc.org.uk/"):
              if any...
                val.txmyType = "FRS-2022"
              else
                val.txmyType = "FRS"
```